pallets/registry/src/lib.rs

```diff
@@ -97,54 +96,83 @@ pub mod pallet {
+        /// data from it.
+        /// 
+        /// This origin must be root until further mechanisms for adding feeds is introduced such as 
+        /// Parachains adding feeds and putting up a staked amount as an economic incentive to avoid
+        /// providing bad data.
         #[pallet::call_index(0)]
         #[pallet::weight(Weight::from_ref_time(10_000) + T::DbWeight::get().reads_writes(1,1))]
-        pub fn reg_feed(origin: OriginFor<T>, key: RegStrT, url: RegStrT) -> DispatchResult {
+        pub fn register_feed(
+            origin: OriginFor<T>,
+            key: RegistryFeedKey<T>,
+            url: RegistryFeedUrl<T>,
+            path: RegistryFeedPath<T>,
+        ) -> DispatchResult {
             let who = ensure_signed(origin)?;
 
             let block_number = <frame_system::Pallet<T>>::block_number();
             let feed = ApiFeed {
-                requested_block_number: block_number,
-                url: Some(url),
+                started_at: block_number,
+                url,
+                path,
+                status: ApiFeedStatus::Registered,
             };
-            ApiFeeds::<T>::insert(&who, &key, feed.clone());
 
+            // Insert the feed into storage.
+            <ApiFeeds<T>>::insert(&who, &key, feed.clone());
             // Emit an event.
-            Self::deposit_event(Event::RegFeed {
-                sender: who,
+            Self::deposit_event(Event::FeedRegistered {
+                by: who,
                 key,
                 feed,
             });
+
             Ok(())
         }
 
-        /// An example dispatchable that may throw a custom error.
+        /// Removes a feed from storage.
+        /// 
+        /// The origin must be the same as the creator which first registered the feed. The only
+        /// other scenario which would cause an feed to be removed is having bad data (getting slashed)
+        /// or other uptime metrics (e.g. the feed errors out too many times).
         #[pallet::call_index(1)]
         #[pallet::weight(Weight::from_ref_time(10_000) + T::DbWeight::get().reads_writes(1,1))]
-        pub fn unreg_feed(origin: OriginFor<T>, key: RegStrT) -> DispatchResult {
+        pub fn unregister_feed(origin: OriginFor<T>, key: RegistryFeedKey<T>) -> DispatchResult {
             let who = ensure_signed(origin.clone())?;
 
-            let feed_exists = ApiFeeds::<T>::contains_key(&who, &key);
-            if feed_exists {
-                let feed = Self::api_feeds(&who, &key).unwrap();
+            // Remove the feed from storage if it exists.
+            // This also implicitly checks that the feed is owned by the origin account.
+            if let Some(feed) = <ApiFeeds<T>>::get(&who, &key) {
                 <ApiFeeds<T>>::remove(&who, &key);
-                Self::deposit_event(Event::UnRegFeed {
-                    sender: who,
+                Self::deposit_event(Event::FeedUnregistered {
+                    by: who,
```

Comment:
```suggestion
                    caller: who,
```
I think `caller`  would be nice (in the contract of substrate)